spec/compiler/codegen/sizeof_spec.cr

```diff
@@ -243,4 +243,105 @@ describe "Code gen: sizeof" do
 
       z)).to_i.should eq(16)
   end
+
+  describe "alignof" do
```

Comment:
**thought:** I'm wondering if it would be reasonable to merge the examples for `sizeof` and `alignof`, as well as `instance_sizeof` and `instance_alignof`, respectively. The programs seem to be almost identical (except for the tested method). If we could avoid compiling mostly identical programs multiple times, spec performance would improve.
Of course this might impact the isolation of specs, where ideally every example tests exactly one thing.
I suppose technically it might be possible to isolate the tests, but still base them on the output of the same sample program if it outputs both, `alignof` and `sizeof`.
```cr
# simplified and incomplete example of what that means:
context "class" do
  sizeof, _, alignof = run(<<-CRYSTAL).partition("|")
    class Foo
      def initialize(@x : Int8, @y : Int32, @z : Int16)
      end
    end
    Foo.new(1, 2, 3)
    "#{sizeof(Foo)}|#{alignof(Foo)}"
    CRYSTAL
  it "sizeof" do
    sizeof.to_i.should eq 8  
  end
  it "alignof" do
    alignof.to_i.should eq 4
  end
end
```
Not sure if this would bring much benefit and would be worth the hassle.
A positive effect would be keeping examples for `alignof` and `sizeof` in sync which probably makes sense because they're closely related.
Feel free to completely ignore this brain dump.

Reply:
String interpolation would require the prelude, which is the opposite of improving spec performance. (In this particular case the `sizeof` and `alignof` are constant expressions, so the whole interpolation is turned into a literal before codegen due to #14075, and the prelude won't actually be needed, but for other types this is not necessarily the case. This is also why I objected to that PR)

The simplest prelude-free way I could think of is:

```crystal
result = run <<-CRYSTAL
  class Box(T)
    def to_unsafe
      pointerof(@object)
    end

    def initialize(@object : T)
    end
  end

  class Foo
    def initialize(@x : Int8, @y : Int32, @z : Int16)
    end
  end

  Foo.new(1, 2, 3)

  Box.new({
    sizeof(Foo),
    alignof(Foo),
    instance_sizeof(Foo),
    instance_alignof(Foo),
  }).to_unsafe
  CRYSTAL

value = result.as(LLVM::GenericValue).to_pointer.as({Int32, Int32, Int32, Int32}*).value
value.should eq({8, 8, 16, 4})
```

Reply:
I don't think https://github.com/crystal-lang/crystal/pull/14075 is the correct reference here. Did you mean #12671?

Anyway, #14090 seems like a good idea which I suppose sparked from this use case.
There's no reason to hold up this PR, of course.